packages/@react-native-windows/cli/src/index.ts

```diff
@@ -58,23 +65,62 @@ export async function generateWindows(
   ns: string,
   options: GenerateOptions,
 ) {
-  if (!fs.existsSync(projectDir)) {
-    fs.mkdirSync(projectDir);
-  }
+  let error;
+  try {
+    if (!fs.existsSync(projectDir)) {
+      fs.mkdirSync(projectDir);
+    }
+
+    installDependencies(options);
 
-  installDependencies(options);
+    const rnwPackage = path.dirname(
+      require.resolve('react-native-windows/package.json', {
+        paths: [projectDir],
+      }),
+    );
+    const templateRoot = path.join(rnwPackage, 'template');
+    await copyProjectTemplateAndReplace(
+      templateRoot,
+      projectDir,
+      name,
+      ns,
+      options,
+    );
+  } catch (e) {
+    error = e;
+    throw e;
+  } finally {
+    if (!options.noTelemetry && !process.env.AGENT_NAME) {
+      const cwd = process.cwd();
+      const pkgJsonPath = findUp.sync('package.json', {cwd});
+      let rnVersion = '';
+      if (pkgJsonPath) {
+        const pkgJson = require(pkgJsonPath);
+        // check how react-native is installed
+        if (
+          'dependencies' in pkgJson &&
+          'react-native' in pkgJson.dependencies
+        ) {
+          // regular dependency (probably an app), inject into json and run install
+          rnVersion = pkgJson.dependencies['react-native'];
+        }
+      }
 
-  const rnwPackage = path.dirname(
-    require.resolve('react-native-windows/package.json', {paths: [projectDir]}),
-  );
-  const templateRoot = path.join(rnwPackage, 'template');
-  await copyProjectTemplateAndReplace(
-    templateRoot,
-    projectDir,
-    name,
-    ns,
-    options,
-  );
+      const rnwPkgJson = require('../package.json');
+
+      telClient.trackEvent({
+        name: 'generate-windows',
+        properties: {
+          error: error,
+          ...options,
```

Comment:
We should probably filter with a known list of flags, or at least sanitize this list ...

Technically `nugetTestFeed` can contain credentials and therefore PII which should never be sent to telemetry.

Reply:
good idea